### PR TITLE
[Intel MKL] Enabling BFloat16 versions of Convolutions, Concat and Id…

### PR DESCRIPTION
**Pull request by nhasabni**
…entity - Round 3

This PR enables BFloat16 versions of 2D and 3D Convolution and their gradient
operators, Concat, and Identity operations.

### CODE REVIEW

File: tensorflow/core/kernels/mkl_conv_ops.cc [hunk 1]
```diff
@@ -1071,17 +1067,19 @@ class MklConvOp : public OpKernel {
...
       std::shared_ptr<ConvFwdPd> conv_fwd_pd = conv_fwd->GetPrimitiveDesc();
       AllocateOutputTensor(context, *conv_fwd_pd, dst_dims_mkl_order, tf_fmt,
                            &dst_tensor);
       Tensor* filter_out_tensor = nullptr;
-      if (typeid(Tinput) == typeid(float) && typeid(Tfilter) == typeid(float) &&
-          typeid(Toutput) == typeid(float)) {
+      if (typeid(Tinput) == typeid(Tfilter) &&
+          typeid(Tinput) == typeid(Toutput) &&
+          (typeid(Tinput) == typeid(float) ||
+           typeid(Tinput) == typeid(bfloat16))) {
```
Comment (penpornk) [inline tensorflow/core/kernels/mkl_conv_ops.cc]: This set of conditions appears twice. (The first time in line 975.) How about storing it in a bool?

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for your PR! One minor change, please.

File: tensorflow/core/kernels/mkl_conv_ops.cc (hunk 1 above)
Comment (nhasabni) [reply tensorflow/core/kernels/mkl_conv_ops.cc]: Sure. Addressed.

**Review by penpornk — APPROVED** [review]
Thank you again for contributing!

**Pull request merged**